sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -607,12 +607,18 @@ def _schedule_periodic_shutdown(self):
     # type: () -> None
     def shutdown_inactive_bundle_processors():
       # type: () -> None
+      inactive_descriptor_ids = []
+      cur_time = time.time()
       with self._lock:
-        for descriptor_id, last_access_time in self.last_access_times.items():
-          if (time.time() - last_access_time >
+        # Shutdown can be expensive, keep out of lock
+        for descriptor_id, last_access_time in access_times:
```

Comment:
```suggestion
        for descriptor_id, last_access_time in self.last_access_times.items():
```

Reply:
Good catch got lost in my second round of changes